### ISSUE TITLE
503 error using v3 endpoint

### ISSUE THREAD

**Issue by onwingslikeeagles** [eduardoboucas/staticman#307]
Hello,

Our team has been using v3 endpoint of the application for a while now and we're experiencing a 503 error. This normally infers an issue with the application's end, so I'm pinging @eduardoboucas for assistance on getting the application back up to speed, if needed.

**Comment by casually-creative** [comment]
Working on the same project, as unpreceden7ed, giving a screenshot of the error. Seems to be CORS related.

![image](https://user-images.githubusercontent.com/5587999/61720483-2675c580-ad67-11e9-8380-fb25115199b8.png)

**Comment by casually-creative** [comment]
I want to make a statement to our progress.

We moved away from the public instance because it's not working anymore, as what this issue is about. We have set up our own instance on heroku and it's working fine as of now. However, we didn't clone this repository because apparently, it's in a broken state. Instead, we cloned the [deploy branch](https://github.com/VincentTam/staticman) of @VincentTam's fork. He went through the effort of merging PRs which brought the repo back to a working state, as explained in #299. Thank you Vincent for that!

So to recap: the public instance is broken because of a CORS / internal issue, the self-hosted instance is broken because of legacy functionalities not working because of v3.

Pinging @eduardoboucas to make you aware of this situation.

**Comment by dancwilliams** [comment]
Thanks for the great info.  Looks like I need to spin up a new instance for myself.  Always fun.

**Comment by VincentTam** [comment]
I also hit this error by using the default API endpoint in the theme [Hugo Future Imperfect Slim](//github.com/pacollins/hugo-future-imperfect-slim).

![Screenshot_2019-08-16_23-15-28](https://user-images.githubusercontent.com/5748535/63198738-f1197a80-c07b-11e9-8aa7-54b94c1eff64.png)

Needs to try other instances:

- @staticmanaima
- @BloggerBust-bot
- ???

**Comment by alexwaibel** [comment]
I've picked up this project. The staticman public instance is not functional right now and your best bet is rolling back to a previous commit you know was working before and self hosting.

**Comment by alexwaibel** [comment]
Duplicate of #317

**Comment by RyanTG** [comment]
@VincentTam I still get this error, using staticman master ("deploy to heroku"), and Hugo Future Imperfect Slim. I assumed that this master branch would have resolved this at this point (based on #319 and such). Do you have any tips?

**Comment by VincentTam** [comment]
> @VincentTam I still get this error, using staticman master ("deploy to heroku"), and Hugo Future Imperfect Slim. I assumed that this master branch would have resolved this at this point (based on #319 and such). Do you have any tips?

@RyanTG I'm sorry I'm now occupied with my job training & I'm learning C#.  I haven't followed the development of Staticman for a year.  I might retake this if I've free time again.

**Comment by RyanTG** [comment]
No problem, good luck with your training. I opted to use staticmanlab, and that's been working for me. But, I question if that's a sustainable approach, given that you're not actively involved. But at any rate, thanks for that tool!

**Comment by VincentTam** [comment]
> No problem, good luck with your training. I opted to use staticmanlab, and that's been working for me. But, I question if that's a sustainable approach, given that you're not actively involved. But at any rate, thanks for that tool!

I'll leave this question to others, while trying to find time to pick up Node.JS (maybe several months) later.

**Comment by alexwaibel** [comment]
@RyanTG when did you deploy Staticman to Heroku? Prior to about 9 days ago when #346 was merged, master was still broken. As of right now, deploying master to Heroku should be working. As far as sustainable approaches, this project currently in need of maintainers. I do my best to resolve any breaking issues as they pop up but do not have much time to actively work on Staticman development lately.

**Comment by RyanTG** [comment]
Yes, it was within 9 days (7 days ago).

I don't know enough about staticmanlab to know what is different with that setup. But once it worked with that app, I gave up on troubleshooting my 503 error.

**Comment by shaftoe** [comment]
@alexwaibel I can confirm master works on Heroku, or at least the master I forked yesterday: https://github.com/shaftoe/staticman

I might have some time to help out here, where should I start from?

**Comment by alexwaibel** [comment]
> @alexwaibel I can confirm master works on Heroku, or at least the master I forked yesterday: https://github.com/shaftoe/staticman
> 
> I might have some time to help out here, where should I start from?

I'd greatly appreciate your contributions @shaftoe! The highest-value item right now is definitely updating [the docs on the website](https://github.com/eduardoboucas/staticman.net). They're super out of date at the moment so it would help a lot of new users if we added things like notes about setting up Heroku instances and whatnot. Should be a good starter task too getting used to the whole system. Right now we have some great articles that I've been pointing users to, but having this info on the site would be better.

Relevant links:
* https://travisdowns.github.io/blog/2020/02/05/now-with-comments.html
* https://spinningnumbers.org/a/staticman-heroku.html
* https://mademistakes.com/articles/improving-jekyll-static-comments/

**Comment by RyanTG** [comment]
I wonder if my issues were theme-related. I read all those relevant links when setting it up. I'll give it a shot with another theme, not https://github.com/pacollins/hugo-future-imperfect-slim, just to rule that out.
